src/Cassandra.IntegrationTests/Linq/LinqMethods/GroupByTests.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+using Cassandra.Mapping;
+using Cassandra.Data.Linq;
+using System.Linq;
+using Cassandra.IntegrationTests.TestBase;
+using Cassandra.Tests.Mapping.Pocos;
+using NUnit.Framework;
+
+namespace Cassandra.IntegrationTests.Linq.LinqMethods
+{
+    [Category("short"), TestCassandraVersion(3, 10)]
+    public class GroupByTests : SharedClusterTest
+    {
+        protected override string[] SetupQueries
+        {
+            get
+            {
+                return new []
+                {
+                    "CREATE TABLE sensor_data (sensor_id text, bucket text, timestamp timeuuid, value double," +
+                    " PRIMARY KEY ((sensor_id, bucket), timestamp))",
+                    "INSERT INTO sensor_data (sensor_id, bucket, timestamp, value) VALUES ('sensor1', 'bucket1', now(), 1.5)",
+                    "INSERT INTO sensor_data (sensor_id, bucket, timestamp, value) VALUES ('sensor1', 'bucket1', now(), 2)",
+                    "INSERT INTO sensor_data (sensor_id, bucket, timestamp, value) VALUES ('sensor1', 'bucket1', now(), 2.5)"
+                };
+            }
+        }
+
+        private static MappingConfiguration GetSensorDataMappingConfig()
+        {
+            return new MappingConfiguration().Define(new Map<SensorData>()
+                .ExplicitColumns()
+                .Column(t => t.Id, cm => cm.WithName("sensor_id"))
+                .Column(t => t.Bucket)
+                .Column(t => t.Timestamp)
+                .Column(t => t.Value)
+                .PartitionKey(t => t.Id, t => t.Bucket)
+                .ClusteringKey(t => t.Timestamp)
+                .TableName("sensor_data"));
+        }
+
+        [Test]
+        public void Should_Project_To_New_Anonymous_Type()
+        {
+            var table = new Table<SensorData>(Session, GetSensorDataMappingConfig());
+            var linqQuery = table
+                .GroupBy(t => new {t.Id, t.Bucket})
+                .Select(g => new
+                {
+                    g.Key.Id,
+                    Avg = g.Average(i => i.Value),
+                    Count = g.Count(),
+                    Sum = g.Sum(i => i.Value),
+                    Min = g.Min(i => i.Value),
+                    Max = g.Max(i => i.Value),
+                    g.Key.Bucket
+                });
+            var results = linqQuery.Execute().ToArray();
+            Assert.AreEqual(1, results.Length);
+            var aggregation = results[0];
+            Assert.AreEqual("sensor1", aggregation.Id);
+            Assert.AreEqual("bucket1", aggregation.Bucket);
+            Assert.AreEqual(aggregation.Avg, 2);
```

Comment:
Use the expected value as first parameter of 
`Assert.AreEqual( int expected, int actual );`